src/utils.py

```diff
@@ -27,5 +28,44 @@ async def hedgedoc_login(aiohttp_client: aiohttp.ClientSession) -> None:
     await aiohttp_client.post("https://md.redbrick.dcu.ie/auth/ldap", data=data)
 
 
+async def get_md_content(url: str, aiohttp_client: aiohttp.ClientSession) -> str:
+    """
+    Get the content of a note at a HedgeDoc URL.
+    """
+    if "https://md.redbrick.dcu.ie" not in url:
+        raise ValueError(f"`{url}` is not a valid MD URL. Please provide a valid URL.")
+    await hedgedoc_login(aiohttp_client)
+    parsed_url = urlparse(url)
+    request_url = (
+        f"{parsed_url.scheme}://{parsed_url.hostname}{parsed_url.path}/download"
+    )
+    async with aiohttp_client.get(request_url) as response:
+        if response.status != 200:
+            raise Exception(
+                f"Failed to fetch the minutes. Status code: `{response.status}`"
```

Comment:
I'd rather the aiohttp exception be raised with `response.raise_for_status()`. Then the specific `aiohttp.ClientResponseError` can be used in the except block instead of the generic `Exception`.